transformer_lens/HookedTransformer.py

```diff
@@ -1462,25 +1469,41 @@ def fold_layer_norm(self, state_dict: Dict[str, torch.Tensor]):
 
         Args:
             state_dict (Dict[str, torch.Tensor]): State dict of pretrained model.
+            :param fold_biases: Enables folding of LN biases. Should be disabled when RMS Norm is used.
+            :param center_weights: Enables the centering of weights after folding in LN. Should be disabled when RMS Norm is used.
         """
         for l in range(self.cfg.n_layers):
             # Fold ln1 into attention - it's important to fold biases first, since biases depend on
             # weights but not vice versa The various indexing is just to broadcast ln.b and ln.w
             # along every axis other than d_model. Each weight matrix right multiplies. To fold in
             # the bias, we use the W_ matrix to map it to the hidden space of the layer, so we need
             # to sum along axis -2, which is the residual stream space axis.
-            state_dict[f"blocks.{l}.attn.b_Q"] = state_dict[f"blocks.{l}.attn.b_Q"] + (
-                state_dict[f"blocks.{l}.attn.W_Q"]
-                * state_dict[f"blocks.{l}.ln1.b"][None, :, None]
-            ).sum(-2)
-            state_dict[f"blocks.{l}.attn.b_K"] = state_dict[f"blocks.{l}.attn.b_K"] + (
-                state_dict[f"blocks.{l}.attn.W_K"]
-                * state_dict[f"blocks.{l}.ln1.b"][None, :, None]
-            ).sum(-2)
-            state_dict[f"blocks.{l}.attn.b_V"] = state_dict[f"blocks.{l}.attn.b_V"] + (
-                state_dict[f"blocks.{l}.attn.W_V"]
-                * state_dict[f"blocks.{l}.ln1.b"][None, :, None]
-            ).sum(-2)
+            if fold_biases:
+                state_dict[f"blocks.{l}.attn.b_Q"] = state_dict[
+                    f"blocks.{l}.attn.b_Q"
+                ] + (
+                    state_dict[f"blocks.{l}.attn.W_Q"]
+                    * state_dict[f"blocks.{l}.ln1.b"][None, :, None]
+                ).sum(
+                    -2
```

Comment:
Nit: In my opinion putting the -2 and ) on separate lines is excessive and harms readability

Reply:
Agreed, I think this was the formatter, will fix

Reply:
Ok it looks like the formatter ci check enforces this style, so I'm going to put it back